packages/renderer/src/lib/preferences/PreferencesResourcesRenderingCopyButton.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+import { onMount } from 'svelte';
+import Fa from 'svelte-fa';
+import { faPaste } from '@fortawesome/free-solid-svg-icons';
+import Tooltip from '../ui/Tooltip.svelte';
+
+let url: string | undefined;
+
+export let path: string;
+
+onMount(async () => {
+  try {
+    const platform = await window.getOsPlatform();
+
+    let prefix = '';
+    if (platform === 'win32') {
+      prefix = 'npipe';
+    } else {
+      prefix = 'unix';
+    }
+
+    url = `${prefix}://${path}`;
+  } catch (e) {
+    console.error('unable to create a socket url from socketPath.', e);
+    url = undefined;
+  }
+});
+
+function copyResourceSocketAddrToClipboard() {
+  window.clipboardWriteText(url!);
+}
+</script>
+
+{#if url}
+  <Tooltip tip="Copy to Clipboard" bottom>
+    <button
+      title="Copy To Clipboard"
+      class="ml-5"
+      aria-label="Copy To Clipboard"
+      on:click="{() => copyResourceSocketAddrToClipboard()}">
+      <Fa icon="{faPaste}" />
+    </button>
+  </Tooltip>
+{/if}
```

Comment:
now you should look at adding a .spec.ts file for this component, you can mimic a lot of existing components in source code have their component being tested using testing-library

Reply:
I developed the test but, it needs 3 ticks to update have the DOM updated. I'm not yet familiar with this test framework and s, can you spot what I'm missing?